packages/teams-js/test/private/store.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { errorLibraryNotInitialized } from '../../src/internal/constants';
+import { store } from '../../src/private/store';
+import { app, AppId } from '../../src/public';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../../src/public/constants';
+import { _minRuntimeConfigToUninitialize, latestRuntimeApiVersion } from '../../src/public/runtime';
+import { Utils } from '../utils';
+
+describe('store', () => {
+  let utils: Utils = new Utils();
+  beforeEach(() => {
+    utils = new Utils();
+    utils.messages = [];
+  });
+  afterEach(() => {
+    app._uninitialize();
+  });
+
+  describe('isSupported', () => {
+    it('should throw if called before initialization', () => {
+      utils.uninitializeRuntimeConfig();
+      expect(() => store.isSupported()).toThrowError(new Error(errorLibraryNotInitialized));
+    });
+  });
+
+  describe('openStoreExperience', () => {
+    const paramFullStore: store.OpenStoreParams = {
+      dialogType: store.StoreDialogType.fullStore,
+    };
+    const paramAppDetail: store.OpenAppDetailParams = {
+      dialogType: store.StoreDialogType.appDetail,
+      appId: new AppId('1542629c-01b3-4a6d-8f76-1938b779e48d'),
+    };
+    const argsAppDetail = ['appdetail', '1542629c-01b3-4a6d-8f76-1938b779e48d'];
+    const paramAppDetailWithoutId = {
+      dialogType: store.StoreDialogType.appDetail,
+    };
+
+    const allowedContexts = [FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage];
+    Object.values(FrameContexts).forEach((context) => {
+      if (allowedContexts.some((allowedContext) => allowedContext === context)) {
+        it(`should throw error when dialog is not supported in ${context} context`, async () => {
```

Comment:
This should be changed to make sure the function throws if called when store is not supported

Reply:
Also, if you start throwing errors out of the function as I suggested, please add unit tests for that as well

Reply:
updated